storage: TestStoreRangeDownReplicate fails quickly under stress

```
make stress PKG=./storage TESTS=TestStoreRangeDownReplicate

go test -v  -tags '' -i -c ./storage -o ./storage/stress.test
github.com/cockroachdb/cockroach/storage
github.com/cockroachdb/cockroach/storage_test
testmain
cd ./storage && stress -stderr -maxfails 1 ./stress.test -test.run "TestStoreRangeDownReplicate" -test.timeout 2m
255 runs so far, 0 failures, over 5s

I161003 19:08:06.777797 1 rand.go:76  Random seed: -5835408053286116671
I161003 19:08:06.791885 22 storage/engine/rocksdb.go:359  opening in memory rocksdb instance
I161003 19:08:06.854477 22 gossip/gossip.go:234  initial resolvers: []
W161003 19:08:06.854508 22 gossip/gossip.go:1045  no resolvers found; use --join to specify a connected node
I161003 19:08:06.854597 22 gossip/gossip.go:268  NodeID set to 1
I161003 19:08:06.856785 22 gossip/gossip.go:268  NodeID set to 1
I161003 19:08:06.857464 22 storage/store.go:1024  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I161003 19:08:06.857584 22 gossip/gossip.go:282  NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:39579" > attrs:<>
I161003 19:08:06.861267 77 storage/replica_trigger.go:254  [s1,r1:/M{in-ax}] new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900ms following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=1970-01-01 00:00:00 +0000 UTC]
I161003 19:08:06.862970 22 storage/engine/rocksdb.go:359  opening in memory rocksdb instance
I161003 19:08:06.864268 22 gossip/gossip.go:234  initial resolvers: [127.0.0.1:39579]
W161003 19:08:06.864340 22 gossip/gossip.go:1047  no incoming or outgoing connections
I161003 19:08:06.864411 22 gossip/gossip.go:268  NodeID set to 2
I161003 19:08:06.876703 22 gossip/gossip.go:268  NodeID set to 2
I161003 19:08:06.876889 22 storage/store.go:1024  [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I161003 19:08:06.876929 22 gossip/gossip.go:282  NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:42977" > attrs:<>
I161003 19:08:06.878634 22 storage/engine/rocksdb.go:359  opening in memory rocksdb instance
I161003 19:08:06.879459 22 gossip/gossip.go:234  initial resolvers: [127.0.0.1:39579]
W161003 19:08:06.879476 22 gossip/gossip.go:1047  no incoming or outgoing connections
I161003 19:08:06.879516 22 gossip/gossip.go:268  NodeID set to 3
I161003 19:08:06.880896 22 gossip/gossip.go:268  NodeID set to 3
I161003 19:08:06.881060 22 storage/store.go:1024  [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
I161003 19:08:06.881107 22 gossip/gossip.go:282  NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:44662" > attrs:<>
I161003 19:08:06.882673 259 gossip/client.go:125  node 2: started gossip client to 127.0.0.1:39579
I161003 19:08:06.883823 22 storage/engine/rocksdb.go:359  opening in memory rocksdb instance
I161003 19:08:06.884724 22 gossip/gossip.go:234  initial resolvers: [127.0.0.1:39579]
W161003 19:08:06.884741 22 gossip/gossip.go:1047  no incoming or outgoing connections
I161003 19:08:06.884781 22 gossip/gossip.go:268  NodeID set to 4
I161003 19:08:06.898785 22 gossip/gossip.go:268  NodeID set to 4
I161003 19:08:06.901616 22 storage/store.go:1024  [n4,s4]: failed initial metrics computation: [n4,s4]: system config not yet available
I161003 19:08:06.901700 22 gossip/gossip.go:282  NodeDescriptor set to node_id:4 address:<network_field:"tcp" address_field:"127.0.0.1:34504" > attrs:<>
I161003 19:08:06.901824 199 gossip/client.go:125  node 3: started gossip client to 127.0.0.1:39579
I161003 19:08:06.905830 22 storage/engine/rocksdb.go:359  opening in memory rocksdb instance
I161003 19:08:06.906666 22 gossip/gossip.go:234  initial resolvers: [127.0.0.1:39579]
W161003 19:08:06.906684 22 gossip/gossip.go:1047  no incoming or outgoing connections
I161003 19:08:06.906725 22 gossip/gossip.go:268  NodeID set to 5
I161003 19:08:06.908113 22 gossip/gossip.go:268  NodeID set to 5
I161003 19:08:06.908299 22 storage/store.go:1024  [n5,s5]: failed initial metrics computation: [n5,s5]: system config not yet available
I161003 19:08:06.908340 22 gossip/gossip.go:282  NodeDescriptor set to node_id:5 address:<network_field:"tcp" address_field:"127.0.0.1:37671" > attrs:<>
I161003 19:08:06.910550 482 gossip/client.go:125  node 4: started gossip client to 127.0.0.1:39579
I161003 19:08:06.910950 360 gossip/server.go:252  node 1: refusing gossip from node 4 (max 3 conns); forwarding to 5 ({tcp 127.0.0.1:37671})
I161003 19:08:06.911259 482 gossip/client.go:130  node 4: closing client to node 1 (127.0.0.1:39579): received forward from node 1 to 5 (127.0.0.1:37671)
I161003 19:08:06.911457 368 gossip/client.go:125  node 4: started gossip client to 127.0.0.1:37671
I161003 19:08:06.918196 610 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161003 19:08:06.918600 610 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161003 19:08:06.918791 610 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161003 19:08:06.918984 610 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161003 19:08:06.919151 610 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161003 19:08:06.919305 610 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161003 19:08:06.919670 508 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:37671->127.0.0.1:39808: use of closed network connection
I161003 19:08:06.919738 352 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:34504->127.0.0.1:49024: use of closed network connection
I161003 19:08:06.919795 280 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:44662->127.0.0.1:39660: use of closed network connection
I161003 19:08:06.919853 197 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:42977->127.0.0.1:45464: use of closed network connection
I161003 19:08:06.919902 14 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:39579->127.0.0.1:54840: use of closed network connection
I161003 19:08:06.920366 483 gossip/client.go:125  node 5: started gossip client to 127.0.0.1:39579
W161003 19:08:06.920414 483 gossip/infostore.go:295  node unavailable; try another peer
W161003 19:08:06.920438 483 gossip/infostore.go:295  node unavailable; try another peer
W161003 19:08:06.920454 483 gossip/infostore.go:295  node unavailable; try another peer
W161003 19:08:06.920485 483 gossip/infostore.go:295  node unavailable; try another peer
W161003 19:08:06.920503 483 gossip/infostore.go:295  node unavailable; try another peer
W161003 19:08:06.920520 483 gossip/infostore.go:295  node unavailable; try another peer
W161003 19:08:06.920536 483 gossip/infostore.go:295  node unavailable; try another peer
W161003 19:08:06.920552 483 gossip/infostore.go:295  node unavailable; try another peer
W161003 19:08:06.920590 483 gossip/infostore.go:295  node unavailable; try another peer
I161003 19:08:06.920665 610 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161003 19:08:06.921444 610 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161003 19:08:06.921569 610 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161003 19:08:06.921670 610 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
I161003 19:08:06.921770 610 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
--- FAIL: TestStoreRangeDownReplicate (0.14s)
panic: [n5,s5]: not connected to gossip [recovered]
	panic: [n5,s5]: not connected to gossip [recovered]
	panic: [n5,s5]: not connected to gossip

goroutine 22 [running]:
panic(0x15ecec0, 0xc420abfbf0)
	/home/jordan/go/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc420184300)
	/home/jordan/go/src/testing/testing.go:579 +0x25d
panic(0x15ecec0, 0xc420abfbf0)
	/home/jordan/go/src/runtime/panic.go:458 +0x243
github.com/cockroachdb/cockroach/util/leaktest.AfterTest.func1()
	/home/jordan/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:73 +0x45c
panic(0x15ecec0, 0xc420abfbf0)
	/home/jordan/go/src/runtime/panic.go:458 +0x243
github.com/cockroachdb/cockroach/storage.(*Store).GossipStore(0xc420694600, 0x7f300aafb108, 0xc420012518, 0x7f300aafb108, 0xc420012518)
	/home/jordan/src/github.com/cockroachdb/cockroach/storage/store.go:1151 +0x456
github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).gossipStores(0xc4204da340)
	/home/jordan/src/github.com/cockroachdb/cockroach/storage/client_test.go:366 +0x13e
github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).initGossipNetwork(0xc4204da340)
	/home/jordan/src/github.com/cockroachdb/cockroach/storage/client_test.go:395 +0x2f
github.com/cockroachdb/cockroach/storage_test.TestStoreRangeDownReplicate(0xc420184300)
	/home/jordan/src/github.com/cockroachdb/cockroach/storage/client_raft_test.go:969 +0xed
testing.tRunner(0xc420184300, 0x19575d0)
	/home/jordan/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/home/jordan/go/src/testing/testing.go:646 +0x2ec


ERROR: exit status 2

558 runs completed, 1 failures, over 10s
FAIL
Makefile:146: recipe for target 'stress' failed
make: *** [stress] Error 1
```

--- a/storage/client_test.go
+++ b/storage/client_test.go
@@ -363,6 +363,7 @@ func (m *multiTestContext) Stop() {
 func (m *multiTestContext) gossipStores() {
 	timestamps := make(map[string]int64)
 	for i := 0; i < len(m.stores); i++ {
+		<-m.gossips[i].Connected
 		if err := m.stores[i].GossipStore(context.Background()); err != nil {
 			m.t.Fatal(err)
 		}

